### ISSUE TITLE
Ticket linked in spec_helper comment is inaccessible

### ISSUE THREAD

**Issue by tobymurray-nanometrics** [rspec/rspec-core#2296]
This appears in [spec_helper.rb](https://github.com/rspec/rspec-core/blob/master/spec/spec_helper.rb)

```
if RUBY_PLATFORM == 'java'
  # Works around https://jira.codehaus.org/browse/JRUBY-5678
  require 'fileutils'
  ENV['TMPDIR'] = File.expand_path('../../tmp', __FILE__)
  FileUtils.mkdir_p(ENV['TMPDIR'])
end
```

Codehaus is totally defunct, so the ticket reference is not helpful. Is this still a needed workaround? Is there an updated reference for what this addresses?

**Comment by myronmarston** [comment]
That originated in #711.  @alindeman, do you remember what the issue was that you worked around?

**Comment by alindeman** [comment]
Hah, vaguely! The tl;dr is that `Dir.mktmpdir { ... }` did not work properly on JRuby ~1.6. According to the release notes, the issue was fixed in JRuby 1.7. If RSpec is testing only on >= JRuby 1.7, it should be good to remove.
